client/question/services/SessionHistoryService.js

```diff
@@ -75,6 +75,10 @@ tie.factory('SessionHistoryService', [
        * Adds a new code balloon to the beginning of the list.
        */
       addCodeBalloon: function(submittedCode) {
+        // The number of balloons pending is increased by assumption that
```

Comment:
We increment the number of balloons here, because adding a code balloon implies that a feedback balloon will soon follow.

(Also, conceptually, I recommend doing this only _after_ you've added the code balloon. It makes no practical difference code-wise but is easier for devs to understand, since otherwise both balloons are "pending" at the time of the increment.)